frontend/appflowy_flutter/lib/plugins/database_view/widgets/group/database_group.dart

```diff
@@ -29,31 +35,70 @@ class DatabaseGroupList extends StatelessWidget {
     return BlocProvider(
       create: (context) => DatabaseGroupBloc(
         viewId: viewId,
-        fieldController: fieldController,
+        databaseController: databaseController,
       )..add(const DatabaseGroupEvent.initial()),
       child: BlocBuilder<DatabaseGroupBloc, DatabaseGroupState>(
-        buildWhen: (previous, current) => true,
         builder: (context, state) {
-          final cells = state.fieldInfos.map((fieldInfo) {
-            Widget cell = _GridGroupCell(
-              fieldInfo: fieldInfo,
-              onSelected: () => onDismissed(),
-              key: ValueKey(fieldInfo.id),
-            );
-
-            if (!fieldInfo.canBeGroup) {
-              cell = IgnorePointer(child: Opacity(opacity: 0.3, child: cell));
-            }
-            return cell;
-          }).toList();
+          final showHideUngroupedToggle = state.fieldInfos.any(
+            (field) =>
+                field.canBeGroup &&
+                field.isGroupField &&
+                field.fieldType != FieldType.Checkbox,
+          );
+          final children = [
+            if (showHideUngroupedToggle)
+              SizedBox(
+                height: GridSize.popoverItemHeight,
+                child: Padding(
+                  padding:
+                      const EdgeInsets.symmetric(horizontal: 12, vertical: 4),
+                  child: Row(
+                    children: [
+                      Expanded(
+                        child: FlowyText.medium(
+                          LocaleKeys.board_showUngrouped.tr(),
+                        ),
+                      ),
+                      Toggle(
+                        value: !state.hideUngrouped,
+                        onChanged: (value) =>
+                            databaseController.updateGroupConfiguration(value),
+                        style: ToggleStyle.big,
+                        padding: EdgeInsets.zero,
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            if (showHideUngroupedToggle) const TypeOptionSeparator(spacing: 0),
```

Comment:
Maybe rename `showHideUngroupedToggle` to `showGroupVisibleToggle`
```suggestion
            if (showHideUngroupedToggle) ...[
              SizedBox(
                height: GridSize.popoverItemHeight,
                child: Padding(
                  padding:
                      const EdgeInsets.symmetric(horizontal: 12, vertical: 4),
                  child: Row(
                    children: [
                      Expanded(
                        child: FlowyText.medium(
                          LocaleKeys.board_showUngrouped.tr(),
                        ),
                      ),
                      Toggle(
                        value: !state.hideUngrouped,
                        onChanged: (value) =>
                            databaseController.updateGroupConfiguration(value),
                        style: ToggleStyle.big,
                        padding: EdgeInsets.zero,
                      ),
                    ],
                  ),
                ),
              ),
              const TypeOptionSeparator(spacing: 0),
            ],
```